### PR TITLE
Remove Tutorial Text when resetting the level

### LINKED ISSUE #309: Reset level (T) doesn't reset tutorial texts

### PR DESCRIPTION
**Pull request by Vhostym**

### CODE REVIEW

File: Assets/Scripts/Level/TutorialTriggerGroup.cs [hunk 1]
```diff
@@ -52,6 +52,7 @@ void UpdateTriggers()
             trigger.AddGameUI(gameUI);
             trigger.AddPlayerProgress(playerProgress);
         }
+        Instance = this;
```
Comment (ambid17) [inline Assets/Scripts/Level/TutorialTriggerGroup.cs]: why do you need this? It's set up in Awake()

File: Assets/Scripts/Level/TutorialTriggerGroup.cs [hunk 2]
```diff
@@ -61,5 +62,7 @@ public void ResetTriggers()
         {
             trigger.ResetTrigger();
         }
+        string[] blankString = { };
+        gameUI.SetupTutorialTexts(blankString);
```
Comment (ambid17) [inline Assets/Scripts/Level/TutorialTriggerGroup.cs]: could just be `gameUI.SetupTutorialTexts(new string[] {});`

**Review by ambid17 — CHANGES_REQUESTED** [review]

File: Assets/Scripts/Level/TutorialTriggerGroup.cs (hunk 1 above)
Comment (Vhostym) [reply Assets/Scripts/Level/TutorialTriggerGroup.cs]: UpdateTriggers is called multiple times, and the awake version has gameUI, playerProgress and TutorialTriggers[] set to null and 0 respectively. In order for the instance to be usable it needs to be kept current. So this update is required.

File: Assets/Scripts/Level/TutorialTriggerGroup.cs (hunk 2 above)
Comment (Vhostym) [reply Assets/Scripts/Level/TutorialTriggerGroup.cs]: Good point, was messing with declarations a bit to see whether I needed "" or {} and forgot to update.

**Review by ambid17 — APPROVED** [review]

**Pull request merged**